File: src/java/test/java/composing/methods/removeAssignmentsToParameters.java
package composing.methods;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

class removeAssignmentsToParameters {
    @Test
    void updateStock() {
        Stock stock = new Stock(5);
        int newStock = OrderHelper.calculateNewStock(stock, 3);

        assertThat(stock.getNbOfItems()).isEqualTo(2);
        assertThat(newStock).isEqualTo(12);
    }
}

File: src/scala/src/test/scala/composing/methods/removeAssignmentsToParameters.scala
package org.ythirion.refactoring.journey
package composing.methods

import org.scalatest.funsuite.AnyFunSuite

class removeAssignmentsToParameters extends AnyFunSuite {
  test("Update stock") {
    val stock = Stock(5)
    val newStock = OrderHelper.calculateNewStock(stock, 3)
    // Stock stays untouched
    assert(stock.nbOfItems == 5)
    assert(newStock == 12)
  }
}
